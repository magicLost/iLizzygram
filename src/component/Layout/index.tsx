import Button from "@material-ui/core/Button";
import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  header: {
    position: "fixed",
    height: "50px",
    width: "100%",
    color: "white",
  },
});

const Layout = ({ children }) => {
  const classes = useStyles();
  return (
    <>
      <header className={classes.header}>
        <p
          style={{
            display: "block",
            maxWidth: "1000px",
            height: "100%",
            margin: "auto",
            paddingTop: "12px",
            backgroundColor: "black",
            textAlign: "center",
          }}
        >
          Layout
        </p>
      </header>

      <div style={{ maxWidth: "1000px", paddingTop: "65px", margin: "auto" }}>
        {children}
      </div>
    </>
  );
};

export default Layout;
