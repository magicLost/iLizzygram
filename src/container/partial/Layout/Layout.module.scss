.container {
  padding-top: 70px;
  max-width: 1600px;
}
