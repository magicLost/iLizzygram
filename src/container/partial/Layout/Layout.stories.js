import React, { useEffect, useState } from "react";
//import { action } from "@storybook/addon-actions";

import Layout from ".";
//import { mockQueriesData } from "../Header/Header.stories";
//import { cache } from "../../../apolloClient/cache";
/* import {
  showAlert,
  showLoginForm,
} from "../../../apolloClient/cache.controller"; */
import Box from "@material-ui/core/Box";
import PureLogo from "../../../component/Logo/PureLogo";
import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
//import thunk from "redux-thunk";
import { modalReducer, alertReducer } from "./../../../store";
import { authReducer } from "../../../auth";

/* const td = require("testdouble");

const Header = td.replaceEsm("./../Header");
td.when(connect()).thenReturn(() => <p>Hello</p>); */

export default {
  component: Layout,
  title: "Pages/Layout",
  decorators: [],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

//CONFIG REDUX
const reducer = combineReducers({
  auth: authReducer,
  modal: modalReducer,
  alert: alertReducer,
});

const store = createStore(reducer);

export const Default = () => {
  useEffect(() => {
    store.dispatch({
      type: "AUTH",
      user: {
        name: "Sia",
        email: "sia@mail.ru",
        uid: "123elrfjweoj324",
      },
    });
  }, []);
  return (
    <Provider store={store}>
      <Layout Logo={PureLogo}>
        <Box maxWidth="1100px" paddingTop="0px">
          <button onClick={() => showAlert("info", "Hello from alert, fukka")}>
            Show alert
          </button>
        </Box>
      </Layout>
    </Provider>
  );
};
