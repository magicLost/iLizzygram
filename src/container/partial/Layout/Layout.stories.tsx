import React, { useEffect, useState } from "react";
//import { action } from "@storybook/addon-actions";

import Layout from ".";
//import { mockQueriesData } from "../Header/Header.stories";
//import { cache } from "../../../apolloClient/cache";
/* import {
  showAlert,
  showLoginForm,
} from "../../../apolloClient/cache.controller"; */
//import PureLogo from "../../../component/Logo/PureLogo";
import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import { Provider } from "react-redux";
//import thunk from "redux-thunk";
import { modalReducer, alertReducer } from "./../../../store";
import { authReducer } from "../../../auth";

/* const td = require("testdouble");

const Header = td.replaceEsm("./../Header");
td.when(connect()).thenReturn(() => <p>Hello</p>); */

export default {
  component: Layout,
  title: "Pages/Layout",
  decorators: [],
  //decorators: [withKnobs],
  // Our exports that end in "Data" are not stories.
  excludeStories: /.*Data$/,
};

//CONFIG REDUX
const reducer = combineReducers({
  auth: authReducer,
  //modal: modalReducer,
  alert: alertReducer,
});

const store = createStore(reducer);

export const Default = () => {
  useEffect(() => {
    store.dispatch({
      type: "AUTH",
      user: {
        name: "Sia",
        email: "sia@mail.ru",
        uid: "123elrfjweoj324",
      },
    });
  }, []);
  return (
    <Provider store={store}>
      <Layout>
        <p>Layout stories</p>
      </Layout>
    </Provider>
  );
};
