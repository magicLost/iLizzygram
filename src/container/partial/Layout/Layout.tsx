import React, { useEffect } from "react";
//import Head from "next/head";
//import Link from "next/link";
import Container from "@material-ui/core/Container";
//import Footer from "../Footer/Footer";
import Header from "../Header/Header";
import classes from "./Layout.module.scss";
//import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
//import Button from "@material-ui/core/Button";
import CenteredTransitionModal from "../../../component/Modal/CenteredTransitionModal/CenteredTransitionModal";
import LoginForm from "../../forms/LoginForm/LoginForm";
import ModalCloseButton from "../../../component/UI/ModalCloseButton/ModalCloseButton";
import {
  hideLoginForm,
  hideAlert,
} from "./../../../hooks/cache/cache.controller";
import Alert from "../../../component/Alert/Alert";
import { useAlert, useModal } from "../../../hooks/cache";
import { useAuthUser, useLogout } from "../../../hooks/auth/auth";
import { checkSavedUser } from "../../../hooks/auth/auth.helper";

interface LayoutProps {
  children: JSX.Element | JSX.Element[];
}

//const useStyles = makeStyles({});

const Layout = ({ children }: LayoutProps) => {
  //const classes = useStyles();

  const { alertState } = useAlert();

  const { modalState } = useModal();

  const { user, loading } = useAuthUser();

  const { logout } = useLogout();

  useEffect(() => {
    checkSavedUser();
  }, []);

  console.log("[RENDER LAYOUT]");

  return (
    <>
      <Header user={user} loading={loading} logout={logout} />
      <Container component={"main"} className={classes.container} maxWidth="lg">
        {children}
      </Container>

      <CenteredTransitionModal
        isShow={modalState.openLoginForm}
        hideModal={hideLoginForm}
      >
        <LoginForm />
        <ModalCloseButton
          ariaLabel="close login form"
          onClick={hideLoginForm}
        />
      </CenteredTransitionModal>

      <Alert
        type={alertState.type}
        isShow={alertState.isShow}
        message={alertState.message}
        hideAlert={hideAlert}
      />
    </>
  );
};

export default Layout;
