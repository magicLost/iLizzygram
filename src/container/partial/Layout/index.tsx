import React from "react";
import Container from "@material-ui/core/Container";
import Header from "../Header";
import classes from "./Layout.module.scss";
//import IModal from "../../../component/IModal";
//import LoginForm from "../../../auth/form/LoginForm";
import Alert from "../../../component/Alert";
//import { hideLoginFormAC, hideForgetPassFormAC } from "../../../store";
//import { IGlobalState } from "../../../store/types";
//import { connect } from "react-redux";
//import ForgetPassForm from "../../../auth/form/ForgetPassForm";

interface LayoutProps {
  Logo: React.FunctionComponent;
  children: JSX.Element | JSX.Element[];
  //isShowLoginForm?: boolean;
  //isShowForgetPassForm?: boolean;
  //hideLoginForm?: () => void;
  //hideForgetPassForm?: () => void;
}

//const useStyles = makeStyles({});

export const Layout = ({
  Logo,
  children,
}: //isShowLoginForm,
//isShowForgetPassForm,
//hideLoginForm,
//hideForgetPassForm,
LayoutProps) => {
  /* const onSuccessLogin = () => {
    hideLoginForm();
  };

  const onSuccessEmail = () => {
    hideForgetPassForm();
  }; */

  console.log("[RENDER LAYOUT]");

  return (
    <>
      <Header Logo={Logo} />
      <Container component={"main"} className={classes.container} maxWidth="lg">
        {children}
      </Container>

      {/* <IModal open={isShowLoginForm} onClose={hideLoginForm}>
        <LoginForm onSuccessLogin={onSuccessLogin} />
      </IModal>

      <IModal open={isShowForgetPassForm} onClose={hideForgetPassForm}>
        <ForgetPassForm onSuccessEmail={onSuccessEmail} />
      </IModal> */}

      <Alert />
    </>
  );
};

/* const mapStateToProps = (state: IGlobalState) => {
  return {
    isShowLoginForm: state.modal.openLoginForm,
    isShowForgetPassForm: state.modal.openForgetPassForm,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    hideLoginForm: () => dispatch(hideLoginFormAC()),
    hideForgetPassForm: () => dispatch(hideForgetPassFormAC()),
  };
}; 

export default connect(mapStateToProps, mapDispatchToProps)(Layout);*/

export default Layout;
