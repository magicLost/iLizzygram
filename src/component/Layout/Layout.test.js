import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from "@testing-library/react";
import { configure } from "@testing-library/dom";
import "@testing-library/jest-dom/extend-expect";

import Layout from ".";

describe("Layout", () => {
  let _render = null;

  describe("Render and props test", () => {
    beforeEach(() => {
      _render = render(<Layout />);
    });

    afterEach(cleanup);

    describe("Snapshots", () => {
      test("matches snapshot", () => {
        const { baseElement } = _render;
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
