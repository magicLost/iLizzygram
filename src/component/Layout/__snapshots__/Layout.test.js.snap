// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Render and props test Snapshots matches snapshot 1`] = `
<body>
  <div>
    <header
      class="makeStyles-header-1"
    >
      <p
        style="display: block; max-width: 1000px; height: 100%; margin: auto; padding-top: 12px; background-color: black; text-align: center;"
      >
        Layout
      </p>
    </header>
    <div
      style="max-width: 1000px; padding-top: 65px; margin: auto;"
    />
  </div>
</body>
`;
